### PR TITLE
[null-safety] remove implicit casts, intl dependency, format

### PR DESCRIPTION
**Pull request by jonahwilliams**
Preparation for null-safety which does not have implicit casts. Removes intl dependency which was only used for a single naming scheme.

Does not bump versioning or changelog

### CODE REVIEW

File: lib/src/record_replay/manifest.dart [hunk 1]
```diff
@@ -47,25 +45,28 @@ class Manifest {
   /// [toJson]), a [FormatException] will be thrown.
   factory Manifest.fromJson(String json) {
     List<Map<String, dynamic>> decoded =
-        new JsonDecoder().convert(json).cast<Map<String, dynamic>>();
-    Manifest manifest = new Manifest();
+        (JsonDecoder().convert(json) as List<dynamic>)
```
Comment (natebosch) [inline lib/src/record_replay/manifest.dart]: It's not typical to construct a `JsonDecoder` manually.

```suggestion
        (jsonDecode(json) as List<dynamic>)
```

File: lib/src/record_replay/manifest.dart [hunk 2]
```diff
@@ -47,25 +45,28 @@ class Manifest {
...
+            .cast<Map<String, dynamic>>();
+    Manifest manifest = Manifest();
```
Comment (natebosch) [inline lib/src/record_replay/manifest.dart:49]: [optional] If you're doing cleanup anyway, consider removing some of the very clearly redundant local variable types.

File: lib/src/record_replay/run_manifest_entry.dart [hunk 3]
```diff
@@ -32,6 +32,48 @@ Encoding _getEncoding(String encoding) {
 
 /// An entry in the process invocation manifest for running an executable.
 class RunManifestEntry extends ManifestEntry {
+  /// Creates a new manifest entry with the given properties.
```
Comment (natebosch) [inline lib/src/record_replay/run_manifest_entry.dart:35]: This comment is not adding any information. How about removing it?

**Review by tvolkert — APPROVED** [review]

File: lib/src/record_replay/manifest.dart (hunk 2 above)
Comment (jonahwilliams) [reply lib/src/record_replay/manifest.dart:49]: always_specify_types 🙃

**Comment by tvolkert** [comment]
Can you bump the version and update the CHANGELOG as well?

**Comment by jonahwilliams** [comment]
@tvolkert I have another commit ready that ops into null safety. I'll update the changelog but I don't really want to release two back to back

**Comment by tvolkert** [comment]
got it - you want me to merge?

**Comment by jonahwilliams** [comment]
Changelog/pubspec updated - ready when you are

**Pull request merged**